### ISSUE TITLE
Unexpected Behaviour from Offset/Repeat Filters

### ISSUE THREAD

**Issue by crantila** [ELVIS-Project/vis-framework#334]
TL;DR: this is the bug caused by some voices with activity and others without.

After running the repeat and offset indexers, then sticking everything into a DataFrame, you end up with a situation where some voices will have lots of `NaN`s where they had repeated identical events that were removed. While the FilterByRepeat removes those `NaN`s, some of them can return when there are events in other voices during the one part's sustained events.

Personally I like this, because it means a composer made the conscious decision to have repeated counterpoint even when another voice was changing. But I have an idea that the profs don't like it, and @alexandermorgan definitely doesn't like it. So we could have a setting.
